### PR TITLE
bugfix parse build_number

### PR DESCRIPTION
**Pull request by jabielecki**
Parse build_number also for alternative log_url format
Closes-Jira-Bug: JD-634

### CODE REVIEW

File: acid/features/history/model.py [hunk 1]
```diff
@@ -106,7 +106,12 @@ def build_number(self):
         try:
             return int(self.log_url.split('/')[-3])
```
Comment (kravvcuCL) [inline acid/features/history/model.py]: I'd suggest:

`
split_url = self.log_url.split('/')

build_number = int(split_url[-3]) if split_url[-3].isdigit() else int(split_url[-4])

return build_number
`

Comment (jabielecki) [reply acid/features/history/model.py]: Changed to be more readable, although without isdigit(), please review.

Comment (jabielecki) [reply acid/features/history/model.py]: Done with isdigit(), sigh.

File: acid/features/history/tests/test_model.py [hunk 2]
```diff
@@ -84,6 +85,23 @@ def test_build_number_return_none_if_no_build_number(self, make_buildset):
...
+            None),
+        ("http://logs.opencontrail.org/p/g/b/664/j/",
+            664)
+    ])
+    @db_session
+    def test_build_number_parsing(self, log_url, expected):
+        build = ZuulBuild(log_url=log_url)
+        assert build.build_number == expected
+
+    # TODO: analyze
```
Comment (xbend) [inline acid/features/history/tests/test_model.py]: Do not smuggle changes irrelevant to the bugfix in this commit.
Separate this into a different commit. It can still be part of the pull request but as a different commit.

File: acid/features/history/model.py [hunk 3]
```diff
@@ -104,7 +104,13 @@ class ZuulBuild(db.Entity):
     @property
     def build_number(self):
         try:
-            return int(self.log_url.split('/')[-3])
+            split_url = self.log_url.split('/')
```
Comment (xbend) [inline acid/features/history/model.py]: Since this code involves some weird 'logic', Id suggest to separate this into a private method like: `def _get_build_number_from_log_url`. In this property just return the returned value from that method.
Inside the method itself, put all the comments explaining what it does, and why it does it, and also why it does it just like its being done here - splitting the url.

Side note, this seems like a configuration thing: which part of the url to take as a build number. Maybe consider putting this into config?

File: acid/features/history/tests/test_model.py (hunk 2 above)
Comment (kam193) [reply acid/features/history/tests/test_model.py]: Maybe it's not necessary when we use squash&merge here, but nice to note this in commit message.

File: acid/features/history/model.py (hunk 3 above)
Comment (kam193) [reply acid/features/history/model.py]: I think separate to private method is the best solution for now

File: acid/features/history/tests/test_model.py (hunk 2 above)
Comment (jabielecki) [reply acid/features/history/tests/test_model.py]: Kamil, I've split skipping the test to another commit. Your approval is crucial due to this project's settings (2 approvals out of 4 members needed).
